iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -38,142 +42,71 @@ class PointerRepository
         ptr_t endPtr{nullptr};
     };
 
-    static constexpr size_t MAX_ID = CAPACITY - 1u;
-    static constexpr size_t MIN_ID = 1u;
-    // remark: 0 is a special purpose id and reserved
-    // id 0 is reserved to interpret the offset just as a raw pointer,
-    // i.e. its corresponding base ptr is 0
+    /// @note 0 is a special purpose id and reserved
+    /// id 0 is reserved to interpret the offset just as a raw pointer,
+    /// i.e. its corresponding base ptr is 0
+    static constexpr size_t MIN_ID = 1U;
+    static constexpr size_t MAX_ID = CAPACITY - 1U;
 
   public:
     static constexpr id_t INVALID_ID = std::numeric_limits<id_t>::max();
 
-    PointerRepository()
-        : m_info(CAPACITY)
-    {
-    }
+    /// @brief default constructor
+    PointerRepository() noexcept;
 
-    bool registerPtr(id_t id, ptr_t ptr, uint64_t size)
-    {
-        if (id > MAX_ID)
-        {
-            return false;
-        }
-        if (m_info[id].basePtr == nullptr)
-        {
-            m_info[id].basePtr = ptr;
-            m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-            if (id > m_maxRegistered)
-            {
-                m_maxRegistered = id;
-            }
-            return true;
-        }
-        return false;
-    }
-
-    id_t registerPtr(const ptr_t ptr, uint64_t size = 0u)
-    {
-        for (id_t id = 1u; id <= MAX_ID; ++id)
-        {
-            if (m_info[id].basePtr == nullptr)
-            {
-                m_info[id].basePtr = ptr;
-                m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-                if (id > m_maxRegistered)
-                {
-                    m_maxRegistered = id;
-                }
-                return id;
-            }
-        }
-
-        return INVALID_ID;
-    }
-
-    bool unregisterPtr(id_t id)
-    {
-        if (id <= MAX_ID && id >= MIN_ID)
-        {
-            if (m_info[id].basePtr != nullptr)
-            {
-                m_info[id].basePtr = nullptr;
+    /// @brief registers the start pointer of the segment in another application with a specific id
```

Comment:
Strictly speaking it does not have to be in another application, but it is used in this way and exists for this very reason. Leave as is for clarity of the use case.